rust-algo/algo/src/map.rs

```diff
@@ -514,22 +519,26 @@ impl<'a> StateTile<'a> {
 
         match unit_type {
             SpawnableUnitType::Firewall(unit_type) => {
-                let unit = Unit {
-                    unit_type,
-                    stability: 1.0,
-                    id: None,
-                    owner: PlayerId::Player1
-                };
-                *self.builder.map.walls.get_mut(coords).unwrap() = Some(unit);
+                if let UnitInformation::Wall{stability,..} = self.builder.atlas.type_info(unit_type.into()) {
```

Comment:
I'd rather have some spaces here ,so it conforms to the style of the rest of the project.

Something like `UnitInformation::Wall { stability , .. }`